src/question/views.py

```diff
@@ -1,3 +1,72 @@
-from django.shortcuts import render
+from rest_framework import viewsets
+from rest_framework import mixins
+from rest_framework import status
+from rest_framework import generics
+from django.shortcuts import get_object_or_404
+from django.http import HttpResponse, JsonResponse
 
-# Create your views here.
+from . import models
+from . import serializers
+from account.models import Profile
+
+
+def handle_vote(abstract_comment, request):
+    user = request.user
+    user_profile = Profile.objects.get(user=user)
+    upvoters = abstract_comment.votes.upvoters
+    downvoters = abstract_comment.votes.downvoters
+    bad_request = HttpResponse(status=status.HTTP_400_BAD_REQUEST)
+    try:
+        vote = request.data['user_vote']
+    except KeyError:
+        return bad_request
+    if vote == 'up':
+        if downvoters.filter(user=user).count() > 0:
+            downvoters.remove(user_profile)
+            upvoters.add(user_profile)
+        else:
+            upvoters.add(user_profile)
+    elif vote == 'down':
+        if upvoters.filter(user=user).count() > 0:
+            upvoters.remove(user_profile)
+            downvoters.add(user_profile)
+        else:
+            downvoters.add(user_profile)
+    elif vote == 'none':
+        if upvoters.filter(user=user).count() > 0:
+            upvoters.remove(user_profile)
+        elif downvoters.filter(user=user).count() > 0:
+            downvoters.remove(user_profile)
+    else:
+        return bad_request
+    return HttpResponse(status=status.HTTP_200_OK)
+
+
+class AnswerVoteViewSet(mixins.UpdateModelMixin, viewsets.GenericViewSet):
+    queryset = models.Answer.objects.all()
+    serializer_class = serializers.AnswerSerializer
+
```

Comment:
Only authenticated users should be able to use this endpoint. 

`permission_classes = (IsAuthenticated, )`

Reply:
Fixed in https://github.com/DevTeamSCH/vikoverflow-backend/pull/12/commits/01b536a456a5d593cff71ce02dcf44d2a7b10a32